ethereumj-core/src/main/java/org/ethereum/sharding/processing/consensus/BeaconStateTransition.java

```diff
@@ -55,6 +63,15 @@ public BeaconStateTransition(StateTransition<Dynasty> dynastyTransition,
     public BeaconState applyBlock(Beacon block, BeaconState to) {
 
         CrystallizedState crystallized = to.getCrystallizedState();
+        ActiveState activeState = to.getActiveState();
+
+        List<AttestationRecord> mergedAttestations = new ArrayList<>();
+        mergedAttestations.addAll(activeState.getPendingAttestations());
+        mergedAttestations.addAll(block.getAttestations());
+        block.getAttestations().forEach(at -> beaconAttester.purgeAttestations(at));
```

Comment:
What do you think if we drive this part by a pub/sub mechanism? It would let us avoid tight components coupling.